.ci/e2eKibana.groovy

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env groovy
+
+@Library('apm@current') _
+
+pipeline {
+  agent none
+  environment {
+    REPO = 'kibana'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    GITHUB_CHECK_E2E_TESTS_NAME = 'E2E Tests'
+    PIPELINE_LOG_LEVEL = "INFO"
+  }
+  options {
+    timeout(time: 3, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    disableConcurrentBuilds()
+  }
+  // http://JENKINS_URL/generic-webhook-trigger/invoke
+  // Pull requests events: https://docs.github.com/en/developers/webhooks-and-events/github-event-types#pullrequestevent
+  triggers {
+    GenericTrigger(
+     genericVariables: [
+      [key: 'GT_REPO', value: '$.repository.full_name'],
+      [key: 'GT_BASE_REF', value: '$.pull_request.base.ref'],
+      [key: 'GT_PR', value: '$.issue.number'],
+      [key: 'GT_PR_HEAD_SHA', value: '$.pull_request.head.sha'],
+      [key: 'GT_BODY', value: '$.comment.body'],
+      [key: 'GT_COMMENT_ID', value: '$.comment.id']
+     ],
+    genericHeaderVariables: [
+     [key: 'x-github-event', regexpFilter: 'comment']
+    ],
+     causeString: 'Triggered on comment: $GT_BODY',
+     printContributedVariables: false,
+     printPostContent: false,
+     silentResponse: true,
+     regexpFilterText: '$GT_REPO$GT_BODY',
+     regexpFilterExpression: '^elastic/kibana/run-fleet-e2e-tests$'
+    )
+  }
+  parameters {
+    string(name: 'kibana_pr', defaultValue: "master", description: "PR ID to use to build the Docker image. (e.g 10000)")
+  }
+  stages {
+    stage('Process GitHub Event') {
+      steps {
+        checkPermissions()
+        buildKibanaDockerImage(refspec: getBranch())
+        catchError(buildResult: 'UNSTABLE', message: 'Unable to run e2e tests', stageResult: 'FAILURE') {
+          runE2ETests('fleet')
+        }
+      }
+    }
+  }
+}
+
+def checkPermissions(){
+  if(env.GT_PR){
+    if(!githubPrCheckApproved(changeId: "${env.GT_PR}", org: 'elastic', repo: 'kibana')){
+      error("Only PRs from Elasticians can be tested with Fleet E2E tests")
+    }
+
+    if(!hasCommentAuthorWritePermissions(env.GT_PR, env.GT_COMMENT_ID)){
+      error("Only Elasticians can trigger Fleet E2E tests")
+    }
+  }
+}
+
+def getBranch(){
+  if(env.GT_PR){
+    return "PR/${env.GT_PR}"
+  }
+  
+  return "PR/${params.kibana_pr}"
+}
+
+def getDockerTag(){
+  if(env.GT_PR){
+    return "${env.GT_PR_HEAD_SHA}"
+  }
+
+  // we are going to use the 'pr12345' tag
+  return "pr${params.kibana_pr}"
```

Comment:
The Docker tag generated by the kibanaDockerBuild step builds and pushes the `pr12345` tag